### PR TITLE
improve: draw button's border if it has the same bg color as campaign (SDKCF-4859)

### PR DESCRIPTION
**Pull request by imaginaris**
# Description
Action button should have a border when campaign body has the same background color.

## Links
SDKCF-4859

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [X] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [X] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [ ] I tested non-trivial changes on a real device
- [X] I ran `./gradlew check` without errors

### CODE REVIEW

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt [hunk 1]
```diff
@@ -193,13 +193,17 @@ internal open class InAppMessageBaseView(context: Context, attrs: AttributeSet?)
         }
 
         buttonView.backgroundTintList = ColorStateList.valueOf(bgColor)
-        // Button stroke color equals to button text color.
-        buttonView.strokeColor = ColorStateList.valueOf(textColor)
+        buttonView.strokeColor = resources.getColorStateList(R.color.modal_border_color_light_grey, context.theme)
```
Comment (vincentdelmundo) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt:196]: Why are we using light grey?

**Review by vincentdelmundo — COMMENTED** [review]
Just one question

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt (hunk 1 above)
Comment (imaginaris) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt:196]: That's the color I got from Figma designs

**Review by vincentdelmundo — APPROVED** [review]
LGTM

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt [hunk 2]
```diff
@@ -250,6 +251,39 @@ internal open class InAppMessageBaseView(context: Context, attrs: AttributeSet?)
...
     }
 
+    // Set button's border based on similarity between its background color and campaign's background color.
+    // The method calculates a distance between colors based on the low-cost approximation algorithm
+    // from [](https://www.compuphase.com/cmetric.htm).
+    // The distance value is compared against a threshold constant to decide if the colors
+    // are visually similar or not.
+    @SuppressWarnings("MagicNumber", "LongMethod")
+    @VisibleForTesting
+    internal fun setButtonBorder(
```
Comment (vincentdelmundo) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt:261]: nice

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt [hunk 3]
```diff
@@ -250,6 +251,39 @@ internal open class InAppMessageBaseView(context: Context, attrs: AttributeSet?)
...
+        val dBlue = Color.blue(bgColor) - Color.blue(buttonBackgroundColor)
+        val distance = round(sqrt(
+            (2 + redMean / 256) * dRed * dRed + 4 * dGreen * dGreen + (2 + (255 - redMean) / 256) * dBlue * dBlue)
+        ).toInt()
+
+        if (distance <= 15) {
+            buttonView.strokeWidth =
+                resources.getDimension(R.dimen.modal_button_border_stroke_width).toInt()
+            if (buttonBackgroundColor == Color.WHITE) {
+                val col = resources.getColorStateList(R.color.modal_border_color_light_grey, context.theme)
```
Comment (vincentdelmundo) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt:278]: Unused variable (from SonarQube)

**Review by vincentdelmundo — APPROVED** [review]
Looks good. Just need to fix the sonarqube error.

**Review by leila-gu — APPROVED** [review]
LGTM

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt (hunk 3 above)
Comment (imaginaris) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageBaseView.kt]: fixed

**Pull request merged**